tools/ansible/tasks/wsu/main.yaml

```diff
@@ -93,39 +93,46 @@
         msg: "Bootstrapper error"
       when: '"Bootstrapping completed successfully" not in bootstrap_out.stderr'
 
-    - name: Initial wait for bootstrap CSR
-      pause:
-        seconds: 60
-
-    - name: Check for bootstrap CSR
-      delegate_to: localhost
-      shell: "oc get csr | awk '/system:serviceaccount:openshift-machine-config-operator:node-bootstrapper/ && /Pending/ {print $1}'"
-      register: bootstrap_csrs
-      until: bootstrap_csrs.stdout != ""
-      retries: 2
-      delay: 60
-
-    - name: Approve pending bootstrap CSRs
-      delegate_to: localhost
-      shell: "oc adm certificate approve {{ item }}"
-      with_items: "{{ bootstrap_csrs.stdout_lines }}"
-
-    - name: Initial wait for node CSR
-      pause:
-        seconds: 60
-
-    - name: Wait for node CSR
-      delegate_to: localhost
-      shell: "oc get csr | awk '/system:node:/ && /Pending/ {print $1}'"
-      register: node_csrs
-      until: node_csrs.stdout != ""
-      retries: 2
-      delay: 60
-
-    - name: Approve pending node CSRs
-      delegate_to: localhost
-      shell: "oc adm certificate approve {{ item }}"
-      with_items: "{{ node_csrs.stdout_lines }}"
+    # Making a best effort to approve CSRs. Not failing until the actual `get node` call, in case the CSRs were approved elsewhere
+    - name: Approve CSRs
+      block:
+        - name: Initial wait for bootstrap CSR
```

Comment:
Do we really need this wait? Isn't it better to fold the wait into the checks?

Reply:
Because we can't look for CSRs specifically belonging to a VM, we have to wait so that if there are existing pending CSRs on the cluster, we do not just approve those and miss the CSR for the bootstrapped node

I'll reduce the time for that initial wait.